external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -314,15 +316,26 @@ private boolean emitTupleIfNotEmitted(ConsumerRecord<K, V> record) {
             boolean isScheduled = retryService.isScheduled(msgId);
             if (!isScheduled || retryService.isReady(msgId)) {   // not scheduled <=> never failed (i.e. never emitted) or ready to be retried
                 final List<Object> tuple = kafkaSpoutConfig.getTranslator().apply(record);
-                if (tuple instanceof KafkaTuple) {
-                    collector.emit(((KafkaTuple)tuple).getStream(), tuple, msgId);
-                } else {
-                    collector.emit(tuple, msgId);
-                }
-                emitted.add(msgId);
-                numUncommittedOffsets++;
-                if (isScheduled) { // Was scheduled for retry, now being re-emitted. Remove from schedule.
-                    retryService.remove(msgId);
+                
+                if(consumerAutoCommitMode){
+                	if (tuple instanceof KafkaTuple) {
+                        collector.emit(((KafkaTuple)tuple).getStream(), tuple);
+                    } else {
+                        collector.emit(tuple);
+                    }
```

Comment:
Could we put in a comment saying something like when autocommit is enabled don't bother to track anything.